lib/duniverse_build.ml

```diff
@@ -65,24 +16,30 @@ let build_cache repo =
 
 let download_cache = "--mount=type=cache,target=/home/opam/.opam/download-cache,uid=1000"
 
+let install_opam_tools =
+  let opam_tools_hash = "6c56ab9fedd7b3f6c143cb606a0ea6fe6a384013" in
+  let open Dockerfile in
+     run "opam pin add -n https://github.com/avsm/opam-tools.git#%s" opam_tools_hash
+  @@ run "opam depext -iy opam-tools"
```

Comment:
wouldn't that come with a risk of interference between packages installed in opam and packages install via the duniverse (I'm not too sure how they interact).

I'd suggest to add the following two lines:
```
@@ "cp \"$(opam exec -- sh -c 'command -v opam-tools')\" ~/"
@@ "run opam remove -a opam-tools"
```
then later use the copied binary `~/opam-tools` directly instead of through opam.

Reply:
It does indeed run that risk, but I'm modifying this in opam-tools to not actually have any dependencies.

Note that the contamination of packages isn't actually a problem for a well-formed duniverse though, since there will never be any use of the opam installed libraries, since _all_ packages are resolved from within the duniverse. This is guaranteed by the duniverse pull.

However, it's still good practise to keep the opam environment as clean as possible. I'll have a few more PRs to refine this behaviour as I continue iterating on opam-tools.